packages/hebao_v1/contracts/base/MetaTxModule.sol

```diff
@@ -127,7 +137,14 @@ contract MetaTxModule is BaseModule
         // solium-disable-next-line security/no-call-value
         (bool success,) = address(this).call.value(msg.value)(data);
```

Comment:
Best to change this to 
```
(bool success,) = address(this).call.gas(gasLimit.sub(startGas.sub(gasleft())).value(msg.value)(data);
```
so the function call can't use more gas than allowed. If it uses more `success == false` and the relayer still gets paid.

Reply:
I added a TODO. The change will cause a stack-too-deep error and I didn't manage to fix it.